Barlist/ReportDlg.cpp

```diff
@@ -51,40 +51,61 @@ void CReportDlg::DoDataExchange(CDataExchange* pDX)
 
 
 BEGIN_MESSAGE_MAP(CReportDlg, CDialog)
-   ON_BN_CLICKED(IDC_PRINT, &CReportDlg::OnClickedPrint)
-   ON_WM_DESTROY()
-   ON_WM_SHOWWINDOW()
+    ON_BN_CLICKED(IDC_PRINT, &CReportDlg::OnClickedPrint)
+    ON_BN_CLICKED(IDC_CHECK_QTY_BY_GROUP, &CReportDlg::UpdateReport)
+    ON_WM_DESTROY()
+    ON_WM_SHOWWINDOW()
 END_MESSAGE_MAP()
 
 
+
+
+
 // CReportDlg message handlers
 
+void CReportDlg::UpdateReport()
+{
+    CWnd* pReport = GetDlgItem(IDC_REPORT);
+    pReport->SetFont(&m_Font);
+
+    CBarlistDoc* pDoc = (CBarlistDoc*)EAFGetDocument();
+
+    //define options parameter
+    
+    CButton* pCheckBox = (CButton*)GetDlgItem(IDC_CHECK_QTY_BY_GROUP);
+    int reportOptions = 0;
+
+    if (pCheckBox->GetCheck() == BST_CHECKED) {
+        reportOptions = CReport::ReportOptions::REPORT_TOTAL_AND_GROUP_QUANTITIES;
+    }
+    if (pCheckBox->GetCheck() == BST_UNCHECKED) {
+        reportOptions = CReport::ReportOptions::REPORT_GROUP_QUANTITIES;
```

Comment:
If unchecked, just the total quantities should be reported.

consider using if - else if

Reply:
I've opted to to just have the if checked case.